added a document on experiment tracking using kfp (#87)
aicoe-aiops/data-science-workflows: docs/track-metrics-using-kubeflow.md

> # Experiment tracking using Kubeflow Pipelines

Reviewer: Is there a video link we can embed here also?

Author: video is WIP, can update the PR with the link once ready :)

Author: embedded the unlisted video. Please take a look

> In this guide you will see how you can use kubeflow pipelines to track metrics from different experiments and runs.

Reviewer: ```suggestion
In this guide you will learn how to use kubeflow pipelines to track metrics from different experiments and runs.
```

> [Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/) is a platform for building and deploying scalable machine learning (ML) workflows. It allows us to automate the running of the jupyter notebooks and scripts using a simple workflow. Using Kubeflow pipelines we can automate the running of multiple steps of a ML pipeline like data collection, model training, inference, which need to be performed sequentially or parallel.

Reviewer: ```suggestion
[Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/) is a platform for building and deploying scalable machine learning (ML) workflows. It allows us to automate the running of  jupyter notebooks and scripts using a simple workflow. Using Kubeflow pipelines we can automate the running of multiple steps of a ML pipeline like data collection, model training, inference, and control which steps need to be performed sequentially and which should run in  parallel.
```

> Kubeflow pipelines also allow us to [export metrics](https://www.kubeflow.org/docs/components/pipelines/sdk/pipelines-metrics/) from a running pipeline which can help compare multiple runs for example over multiple model training runs. We can compare the achieved model performance metrics like accuracy, f1 score achieved.

Reviewer: ```suggestion
Kubeflow pipelines also allow us to [export metrics](https://www.kubeflow.org/docs/components/pipelines/sdk/pipelines-metrics/) from a running pipeline. This helps us compare multiple runs over multiple model training runs. We can also compare the achieved model performance metrics like accuracy, f1, etc.
```

## Pre-requisites

> 1. Existing kubeflow pipeline with components (notebooks or scripts) for which you want to track metrics.

Reviewer: ```suggestion
1. Existing kubeflow pipeline instance available and the components (notebooks or scripts) that you want to track metrics for.
```

## Track Metrics

> To enable tracking of metrics during the execution of a machine learning experiment, your component must have an output called `mlpipeline-metrics`  which must return a JSON-serialized metrics dictionary. Let's demonstrate this with the help of this [notebook](https://github.com/os-climate/aicoe-osc-demo/blob/master/notebooks/demo1/demo1-create-tables.ipynb).

Reviewer: Can we remove the link to that notebook here?  Move it to the very bottom of the doc, add something like, "if you want to see an real example of how to integrate experiement tracking, take a look at this noteboook"

Author: sounds good.

> In this notebook, we want to capture metrics such as the time taken to execute certain queries, and the time taken to upload some files to S3 storage.

Reviewer: ```suggestion
In this notebook, we want to capture metrics such as the time taken to execute certain queries, and the time taken to upload some files to S3 storage.
```
```suggestion
Below we will demonstrate how to capture metrics for the time taken to execute certain queries, and the time taken to upload some files to S3 storage.
```
